tests/src/lib.rs

```diff
@@ -207,3 +207,10 @@ async fn ws_more_request_than_buffer_should_not_deadlock() {
 		req.await.unwrap();
 	}
 }
+
+#[tokio::test]
+async fn wss_works() {
```

Comment:
Would be good to be complete here:
- assert that `ws://` works
- assert that omitting the port works (and uses the default)
- assert that providing a wacky port fails, bigger than `u16::MAX`, negative number, `0` etc
- assert that host name must be ascii (does it?)